### PR TITLE
feat: Add search by title functionality

### PR DESCRIPTION
**Pull request by sadakchap**
# Title and Issue number 

Title : [Challenges] Search by Challenge

Issue No. : https://github.com/sadanandpai/frontend-mini-challenges/issues/385

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have created a helpful and easy to understand `README.md`
- [] I have updated the Index.html file for my contribution

### CODE REVIEW

File: shared/data/utils/challenges.helper.ts [hunk 1]
```diff
@@ -11,3 +11,9 @@ export function getSortedChallengesByDifficulty(challenges: Map<string, IChallen
   });
   return sortedChallengesByDifficulty;
 }
+
+//TODO: make filter function that takes key to search in & value to search for, and boolean for case-insensitive
+export function getSortedChallengesByTitle(challenges: Map<string, IChallenge>, value: string) {
```
Comment (sadanandpai) [inline shared/data/utils/challenges.helper.ts:16]: name is confusing.
Function does filtering but name says sorting

File: apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx [hunk 2]
```diff
@@ -1,41 +1,67 @@
...
+        <div className={styles.searchInputWrapper}>
+          <input type="text" name="searchTextInput" placeholder='Search challenge...' className={styles.searchInput} value={searchInput} onChange={(e) => setSearchInput(e.target.value.trim())} />
+          <img src={assets.searchIconSVG} alt="search challenges by title" width={15} height={15} className={styles.searchIcon} />
+        </div>
+        <div className={styles.filterByTechWrapper}>
+          {links.map((link) => <Link to={`/${link.tech}`} key={link.tech}>
+            <img src={link.imgSrc} width={35} height={35} className={link.active ? styles.activeTech : ''} alt={`filter by ${link.tech}`} />
+          </Link>)}
+        </div>
+      </div>
```
Comment (sadanandpai) [inline apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:40]: can we move the above one as a separate component

**Review by sadanandpai — APPROVED** [review]

File: shared/data/utils/challenges.helper.ts (hunk 1 above)
Comment (sadakchap) [reply shared/data/utils/challenges.helper.ts]: https://github.com/sadanandpai/frontend-mini-challenges/pull/396/commits/e4d42435534cbd14bc9fd3c20a83d9641677f091

File: apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx (hunk 2 above)
Comment (sadakchap) [reply apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:40]: the whole filter row? or only filter by tech div? @sadanandpai

Comment (sadanandpai) [reply apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:40]: whole

Comment (sadakchap) [reply apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:40]: I have created another pr, as this was already merged. Please review this [pr](https://github.com/sadanandpai/frontend-mini-challenges/pull/402). 🙂

**Comment by sadakchap** [comment]
Thank you @sadanandpai for reviewing the pr 😇

**Pull request merged**